### ISSUE TITLE
xSchedule Background Sequence Broken in 2024.19

### ISSUE THREAD

**Issue by Clyde-L** [xLightsSequencer/xLights#5021]
xSchedule is not properly activating background sequences.  I've used the same schedule for the past 3 years, only changing the FSEQ files in the play list.  I left my Background sequence alone, just rendering to the new network/layout settings.  Today I created a new Playlist and added a new static sequence (first time in years).  And now, when I play this new Playlist, the static sequence won't play and the background sequence does.  They won't play concurrently.

I reverted to a 2021.32 version of xLights and my schedule started playing correctly; A static sequence Playlist with a Background sequence play list showing my Tune-to info.  

I then updated back to 2024.19, opened xSchedule, and it's back to working again.  
I tried using xLights 2022.22 and it didn't work.  This may have been broken for a few years?

Please see if you can reproduce?

Thank you for looking :)

Clyde L

**Comment by derwin12** [comment]
Can you do a view log, save it then attach it here?

**Comment by Clyde-L** [comment]
> Can you do a view log, save it then attach it here?

no luck.  suggestions?

![image](https://github.com/user-attachments/assets/0f3545e5-20ef-493f-ae7c-6ac1afbf456c)

**Comment by derwin12** [comment]
Go to that file in file explorer and right click compress to a zip and attach that?

**Comment by Clyde-L** [comment]
> Go to that file in file explorer and right click compress to a zip and attach that?



where is this located in the folder?  What is it called?  I see nothing

**Comment by derwin12** [comment]
It was in your screenshot %appdata%\Roaming\xschedule_l44cpp.log

**Comment by Clyde-L** [comment]
Um....
13gb lol

![image](https://github.com/user-attachments/assets/6a72179b-592d-404f-a906-27d22d015cda)

**Comment by Clyde-L** [comment]
Took a while to zip it up, but here is a link you can download it:

https://www.dropbox.com/scl/fi/g2om90c4f8arcy1h9k07r/xSchedule_l4cpp.zip?rlkey=9k9gj8j4b5cte8i1fxu1jq3ll&dl=0

**Comment by derwin12** [comment]
Logfile goes back to 2019 :-)

Anyways .. the log says
Error pre-reading FSEQ file (D:\Dropbox\Dropbox\Pixel Pro Displays\Web Site\Website layout\Pro Layout\Pro Layout Testing for New Models\Rilla\New folder (2)\2023 Tune to Background.fseq), fopen returned NULL

Permissions or the file is empty?
![image](https://github.com/user-attachments/assets/fd9f3aca-4e34-4004-824d-1f0cf51e8b7d)

No harm renaming that file so that a new one is created (with xschedule not running)

**Comment by Clyde-L** [comment]
So you're saying that I should rename the sequence.  Render and save (new FSEQ). Do this when xSchedule is not open?

And yes, that's been my show-only computer since 2017.  Lots of data there for you to go thru.  Love xSchedule.

**Comment by Jmedley82** [comment]
I'm on 2024.15 and and if I have background playlist selected it won't play anything. If I remove background playlist it business as usual.  Logs show playlist starting and stopping. I see no errors in the logs.

**Comment by derwin12** [comment]
No I was referring to just this log file .. get it to cycle to a new one so it is not so huge. Not the sequence.

As for your issue .. I think it is the fseq file - is it perhaps not on your machine but in the dropbox cloud?

I see this one as well. It just cant read these fseq files.
```
2022-12-12 15:55:26,754 8952 log_base [INFO]          ######## Playlist step 2021 Tune to Background stopping.
2022-12-12 15:55:26,754 8952 log_base [INFO]          ######## Playlist step 2021 Tune to Background starting.
2022-12-12 15:55:26,754 8952 log_base [ERROR] Error pre-reading FSEQ file (D:\Dropbox\Dropbox\Nutcracker\2021 Tune to Background.fseq), fopen returned NULL
```

**Comment by Clyde-L** [comment]
> No I was referring to just this log file .. get it to cycle to a new one so it is not so huge. Not the sequence.
> 
> As for your issue .. I think it is the fseq file - is it perhaps not on your machine but in the dropbox cloud?
> 
> I see this one as well. It just cant read these fseq files.
> 
> ```
> 2022-12-12 15:55:26,754 8952 log_base [INFO]          ######## Playlist step 2021 Tune to Background stopping.
> 2022-12-12 15:55:26,754 8952 log_base [INFO]          ######## Playlist step 2021 Tune to Background starting.
> 2022-12-12 15:55:26,754 8952 log_base [ERROR] Error pre-reading FSEQ file (D:\Dropbox\Dropbox\Nutcracker\2021 Tune to Background.fseq), fopen returned NULL
> 


The files are stored locally on my PC and synced to all my other devices.  I only use this PC as a show computer so it is stored on that machine.

As far a regenerating a new file, I am still not exactly sure how to do that.  I must have closed xSchedule 4 times today LOL

**Comment by derwin12** [comment]
close xschedule .. go to that directory in file explorer and just rename it.
Now open xschedule and it will create a new one.

As for your .fseq. Perhaps copy it to your desktop or some other location and repoint it to that location to see if that is the issue.

**Comment by Clyde-L** [comment]
Thank you, I was able to create a new one.  I am having issues now with my show playing at all when a background sequence is active.  As of right now only the background sequence started to play and the static (Pre2024) Playlist doesn't show any programming.  As soon as de-select the background sequence, the static show plays.  

Below is the current log.
Thanks,
Clyde L
[xSchedule Log.txt](https://github.com/user-attachments/files/18015920/xSchedule.Log.txt)

**Comment by derwin12** [comment]
Nothing obvious to me .. Perhaps Keith can take a look?

**Comment by Clyde-L** [comment]
Still not able to get this to work.  Anyone can look into it?

**Comment by dev8edss** [comment]
background playlist doesn't actually play in the background. it is processed after any playing playlist https://youtu.be/7cNYgiKZraw?t=221.

either limit the channels on the background playlist or set the blend mode of the background playlist to "overwrite skip black"

**Comment by Clyde-L** [comment]
Dev8edss, thank you for pointing out the blend mode.  I believe that is the problem.  Changing the blendmode to "Overwrite skip black" did the trick and it appears to be working.  Thank you!

While helping another user, I must have  changed the blend mode to "Overwrite" and recreated what they were experiencing. Personally I ran the  "Overwrite if zero" blend mode and never had a problem. When I tried switching back it seemed to stop working, and I had been running it in  "Overwrite if zero" blend mode since 2020.  After changing to  "Overwrite if zero", the tune-to background resumed working as normal.  Since this happened 2 weeks ago, my fix was to import tune-to effects into every sequence, then apply to the appropriate models and sub models.

Should anyone else happen upon this closed thread I'll add some clarification.  I never sequence on my tune-to signs or submodels of props that have a tune-to sign.  Their models and sub models are not part of any groups or All Display effects.  Effectively they are removed from any programming.  The "Overwrite skip black" works fine to project my TuneTo FSEQ file onto only the Tune-to elements/sub models, while allowing the effects from the static sequence to play (which do not have sequencing on any of the tune-to elements.  I will find time this weekend to open all my sequences, delete the imported effects on my sings, and resume using the background sequence with blendmode set to "Overwrite skip black."
